lib/core/CStaticThreadPool.cc

```diff
@@ -21,11 +22,18 @@ std::size_t computeSize(std::size_t hint) {
 }
 }
 
-CStaticThreadPool::CStaticThreadPool(std::size_t size)
-    : m_Busy{false}, m_Cursor{0}, m_TaskQueues{computeSize(size)} {
-    m_NumberThreadsInUse.store(m_TaskQueues.size());
-    m_Pool.reserve(m_TaskQueues.size());
-    for (std::size_t id = 0; id < m_TaskQueues.size(); ++id) {
+CStaticThreadPool::CStaticThreadPool(std::size_t size, std::size_t queueCapacity)
+    : m_Busy{false}, m_Cursor{0} {
+
+    std::size_t poolSize = computeSize(size);
+    m_Pool.reserve(poolSize);
+
+    for (std::size_t id = 0; id < poolSize; ++id) {
+        m_TaskQueues.push_back(std::make_unique<TWrappedTaskQueue>(queueCapacity));
+    }
```

Comment:
I should have commented on this in the description sorry.

`CConcurrentQueue` not move insertable and it doesn't have a copy ctor either. The unique pointer was the only way I could think of adding the class to the container. 

It worked before because `CConcurrentQueue` had a default no parameter ctor, the queues were initialised by  `m_TaskQueues{computeSize(size)}`. No that no longer works

Reply:
Oh it is mutex which stops it being moveable; that's a pain. We should probably make `CConcurrentQueue` explicitly not moveable, by deleting its move operations. At the moment it looks like it is moveable reading the code.

Reply:
I deleted the move and copy options (rather than inheriting CNonCopyable) and added a comment